pkg/ocp/machinesets.go

```diff
@@ -207,19 +206,28 @@ func (msd *k8sMachineSetDeployer) List() ([]unstructured.Unstructured, error) {
 			return nil, errors.Wrapf(err, "failed to get label from machineset ")
 		}
 
-		if labels[MachineSetOwnerLabel] == MachineSetOwnerSubmariner {
+		if labels[SubmarinerGatewayLabel] == "true" {
 			resultList = append(resultList, machinesetItems[i])
 		}
 	}
 
 	return resultList, nil
 }
 
-func FilterDedicatedNode(gwNodes []v1.Node) []v1.Node {
+func RemoveDuplicates(machineSets []unstructured.Unstructured, gwNodes []v1.Node) []v1.Node {
 	var resultNode []v1.Node
 
 	for i := 0; i < len(gwNodes); i++ {
-		if gwNodes[i].GetLabels()[MachineSetOwnerLabel] != MachineSetOwnerSubmariner {
+		addToResult := true
+
+		for i := 0; i < len(machineSets); i++ {
+			if strings.Contains(gwNodes[i].GetName(), machineSets[i].GetName()) {
```

Comment:
Can we safely assume that machineSet.name will be same as gwNode.Name?

Following is the output from one of the deployments.

```
[sgaddam@localhost submariner]$ oc get nodes
NAME         STATUS   ROLES                  AGE   VERSION
master-0-0   Ready    control-plane,master   19d   v1.25.4+77bec7a
master-0-1   Ready    control-plane,master   19d   v1.25.4+77bec7a
master-0-2   Ready    control-plane,master   19d   v1.25.4+77bec7a
worker-0-0   Ready    worker                 19d   v1.25.4+77bec7a
worker-0-1   Ready    worker                 19d   v1.25.4+77bec7a

```

```
[sgaddam@localhost submariner]$ oc get machinesets -A
NAMESPACE               NAME                     DESIRED   CURRENT   READY   AVAILABLE   AGE
openshift-machine-api   ice01-0-qjgw7-worker-0   2         2         2       2           19d

```

Reply:
I think it should depend on the machineset provider, for Azure and RHOS that is the pattern I observed. We may need an extra label, if we are planning not using the name.

Reply:
Even if its a pattern today, things might change anytime. So, IMHO we should identify alternatives to this.

Please check if there is a way to know the node/machine-name from the machineset - I think there should be one (the machineset controller should be using it to identify the associated machines).

Reply:
Just checked, 

the below one to get info about node from machineset return emty

`% kubectl get machinesets -n openshift-machine-api subgw-eastus-05ea1c -o yaml | grep subgw-eastus-05ea1c-6475l`

and when tried the otherway , it does not have any reference to the machine name.

```
% kubectl get nodes subgw-eastus-05ea1c-6475l -o yaml | grep  subgw-eastus-05ea1c                                         
    cloud.network.openshift.io/egress-ipconfig: '[{"interface":"subgw-eastus-05ea1c-6475l-nic","ifaddr":{"ipv4":"10.0.128.0/17"},"capacity":{"ip":255}}]'
    csi.volume.kubernetes.io/nodeid: '{"disk.csi.azure.com":"subgw-eastus-05ea1c-6475l","file.csi.azure.com":"subgw-eastus-05ea1c-6475l"}'
    machine.openshift.io/machine: openshift-machine-api/subgw-eastus-05ea1c-6475l
    kubernetes.io/hostname: subgw-eastus-05ea1c-6475l
  name: subgw-eastus-05ea1c-6475l
  providerID: azure:///subscriptions/3c7a4fa5-8496-4cdc-8a78-e344e2f896d7/resourceGroups/asuryana1-tbrrj-rg/providers/Microsoft.Compute/virtualMachines/subgw-eastus-05ea1c-6475l
  - address: subgw-eastus-05ea1c-6475l
```